lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.cpp

```diff
@@ -1000,7 +1000,7 @@ bool WiFiClientSecure_light::_connectSSL(const char* hostName) {
     // ============================================================
     // Set the EC Private Key, only USE_MQTT_AWS_IOT
     // limited to P256 curve
-    br_ssl_client_set_single_ec(_sc.get(), _chain_P, 1,
+    br_ssl_client_set_single_ec(_sc.get(), _chain_P, _chain_P ? 1 : 0,

```

Comment:
This is not right. You cannot set a client ec and say it has zero instances. The test must be upwards.  Tasmota shouldn't call this function at all if no EC cert is present

Reply:
The probles here is that if you call once this (br_ssl_client_set_single_ec), then internal vtable is initialized `(cc->client_auth_vtable = &cc->client_auth.single_ec.vtable;` and once it is initialized all calls are done and it may reach code in library (above) once you kill your key you are in trouble

Reply:
The problem is that you shouldn’t call this function if you don’t have an EC client certificate.